trio/_core/_ki.py

```diff
@@ -109,6 +108,14 @@ def currently_ki_protected():
     return ki_protection_enabled(sys._getframe())
 
 
+# This is to support the async_generator package necessary for aclosing on <3.10
+# functions decorated @async_generator are given this magic property that's a
+# reference to the object itself
```

Comment:
I think there are multiple scenarios of continued async_generator dependency by apps and libraries, so this comment may be confusing.

1. not using aclosing, but haven't bothered to migrate off of async_generator.  This is fair, and it may not be under an application's control-- the dependency may be coming in by transitive dependency, like some generator util library.
2. using async_generator for aclosing only (not using the module's generators) - it's perfectly fine, and has no bearing on this legacy detector.

Reply:
Sure, feel free to update it in a separate PR.